src/GitHub.Exports/Helpers/ThreadingHelper.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System.Threading.Tasks;
+using ThreadHelper = Microsoft.VisualStudio.Shell.ThreadHelper;
+using GitHub.Extensions;
+
+namespace GitHub.Helpers
+{
+    public static class ThreadingHelper
+    {
+        public async static Task SwitchToMainThreadAsync()
+        {
+            if (!Guard.InUnitTestRunner())
```

Comment:
Yes, it's pretty insidious when the compiler starts piling on the syntactic sugar! It was good of you to point it out, you never know when it gets missed, this stuff is sneaky :P